apps/desktop/src/components/v3/StackCommitDetails.svelte

```diff
@@ -0,0 +1,63 @@
+<script lang="ts">
+	import StackDetailsCommitHeader from './StackDetailsCommitHeader.svelte';
+	import StackDetailsFileList from './StackDetailsFileList.svelte';
+	import { ProjectService } from '$lib/project/projectService';
+	import { inject } from '@gitbutler/shared/context';
+	import Icon from '@gitbutler/ui/Icon.svelte';
+	import type { Commit, WorkspaceBranch } from '$lib/branches/v3';
+
+	interface Props {
+		stackId: string;
+		selectedCommitId: string | undefined;
+		selectedCommitDetails?: Commit;
+		selectedBranchDetails?: WorkspaceBranch;
+	}
+
+	let {
+		selectedCommitId = $bindable(),
+		stackId,
+		selectedCommitDetails: commit,
+		selectedBranchDetails
+	}: Props = $props();
```

Comment:
Could we name these as follows perhaps?
```ts
	let {
		stackId,
		commit,
		branch
	}: Props = $props();
```

Reply:
Actually, I take that back.. Could we pass them as props and get them from the service? As in, stackid, commitId, and branchId?

Reply:
Yeah I'll refactor this in a follow-up :+1: